File: test-suite/src/test/java/io/micronaut/microstream/docs/CounterService.java
package io.micronaut.microstream.docs;

import io.micronaut.cache.annotation.CacheConfig;
import io.micronaut.cache.annotation.CachePut;
import io.micronaut.cache.annotation.Cacheable;
import jakarta.inject.Singleton;

import java.util.HashMap;
import java.util.Map;

@Singleton
@CacheConfig("counter") // <1>
public class CounterService {

    Map<String, Long> counters = new HashMap<>();

    @Cacheable // <2>
    public Long currentCount(String name) {
        return counters.get(name);
    }

    @CachePut(parameters = {"name"}) // <3>
    public Long setCount(String name, Long count) {
        counters.put(name, count);
        return count;
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/CounterService.kt
package io.micronaut.microstream.docs

import io.micronaut.cache.annotation.CacheConfig
import io.micronaut.cache.annotation.CachePut
import io.micronaut.cache.annotation.Cacheable
import jakarta.inject.Singleton
import java.util.HashMap

@Singleton
@CacheConfig("counter") // <1>
open class CounterService {

    val counters: MutableMap<String, Long> = HashMap()

    @Cacheable // <2>
    open fun currentCount(name: String): Long? = counters[name]

    @CachePut(parameters = ["name"]) // <3>
    open fun setCount(name: String, count: Long): Long {
        counters[name] = count
        return count
    }
}
